### PR TITLE
Do not return shares for files outside "files"

### LINKED ISSUE #15139: File in trash still appears as shared for recipient
### Steps to reproduce
1. Create two users "user1" and "user2"
2. Login as "user1"
3. Create a folder "folder" and subfolder "folder/test"
4. Share "folder/test" with "user2"
5. Delete "folder"
6. Check trashbin: folder is present
7. Check "Shared with others" in the files app
8. Login as "user2"
9. Check if "test" is present

### Expected result
Folder "test" not visible for "user2" any more.
Folder "test" not visible in "Shared with others" section.

### Actual result
Folder "test" is still visible for "user2" until "user1" deletes it from trash.
Folder "test" still visible in "Shared with others" section.

Since the introduction of "moving files to trash" and the trashbin storage wrapper, they now keep their file ids. This is nice because it would make it possible to keep shares associated to trashed files, but the share API must be adapted to not return such entries.

Note that if "folder" was shared directly, the share would be cleant up properly. The reason why it doesn't in this specific case is because of https://github.com/owncloud/core/issues/14664

Discussed with @schiesbn and he said we could do the following:
- [ ] prevent the share API from touching shares linked with trash files
- [ ] remove the clearing of shares on direct delete to make it possible to keep shares associated with trash files

### Versions
The bug happens in 8.0.2 (since the introduction of the trashbin storage wrapper)

### PR DESCRIPTION
**Pull request by PVince81**
Fixes https://github.com/owncloud/core/issues/15139

Please review @icewind1991 @schiesbn @LukasReschke @nickvergessen @MorrisJobke

### CODE REVIEW

**Comment by PVince81** [comment]
Steps here: https://github.com/owncloud/core/issues/15139#issue-63945183 

I think the PR is small enough and can be backported as is (backport confirmation was given here https://github.com/owncloud/core/issues/15139#issuecomment-85450555)

File: lib/private/share/share.php [hunk 1]
```diff
@@ -318,6 +318,7 @@ public static function getItemSharedWithUser($itemType, $itemSource, $user, $own
 			$fileDependend = true;
 			$column = 'file_source';
 			$where = 'INNER JOIN `*PREFIX*filecache` ON `file_source` = `*PREFIX*filecache`.`fileid` WHERE';
+			$where .= ' `*PREFIX*filecache`.`path` LIKE \'files/%\' AND';
```
Comment (icewind1991) [inline lib/private/share/share.php]: breaks shares from external storages

Comment (PVince81) [inline lib/private/share/share.php]: Damn. Too bad we cannot detect ext storages that easily.

Should I replace this with "NOT LIKE 'files_%/%'" ? But then if someone has a folder "files_trashbin" on their ext storage, it won't work either :rage:

Comment (nickvergessen) [inline lib/private/share/share.php]: and is dead slow

**Comment by PVince81** [comment]
As discussed with @icewind1991 and @nickvergessen, will move the check to PHP.
In that part it will check whether the target file is actually inside the user's view (like it's done for favorites)

**Comment by PVince81** [comment]
#### Tests
- [x] Steps from https://github.com/owncloud/core/issues/15139#issue-63945183, share is not visible any more if the target is in trash
- [x] Share still visible for non-trash files/folders
- [x] Share still visible if the shared file/folder is on an external storage

**Comment by icewind1991** [comment]
Looks good, did some basic testing :+1:

**Comment by MorrisJobke** [comment]
Works :+1:

**Comment by MorrisJobke** [comment]
There was a backport request and approval in https://github.com/owncloud/core/issues/15139  

@PVince81 Can you come up with the PR? Thanks

**Comment by jnfrmarks** [comment]
What's the risk in taking this now for 8.0.3?

**Comment by MorrisJobke** [comment]
I added the backport request to not lose focus of this

**Comment by MorrisJobke** [comment]
@jnfrmarks 8.0.3 should be released already :see_no_evil:

**Comment by karlitschek** [comment]
@MorrisJobke Should. But unfortunately it isn´t because of an unresolved DB issue.

**Comment by PVince81** [comment]
stable8 backport PR: https://github.com/owncloud/core/pull/15609

**Pull request merged**